homeassistant/components/tplink_lte.py

```diff
@@ -0,0 +1,103 @@
+"""
+Support for TP-Link LTE modems.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/tplink_lte/
+"""
+import asyncio
+from datetime import timedelta
+import logging
+
+import voluptuous as vol
+import attr
+import aiohttp
+
+from homeassistant.const import (
+    CONF_HOST, CONF_PASSWORD, EVENT_HOMEASSISTANT_STOP)
+from homeassistant.helpers import config_validation as cv
+from homeassistant.helpers.aiohttp_client import async_create_clientsession
+
+REQUIREMENTS = ['tp-connected==0.0.4']
+
+_LOGGER = logging.getLogger(__name__)
+
+MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=10)
+
+DOMAIN = 'tplink_lte'
+DATA_KEY = 'tplink_lte'
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.All(cv.ensure_list, [vol.Schema({
+        vol.Required(CONF_HOST): cv.string,
+        vol.Required(CONF_PASSWORD): cv.string,
+    })])
+}, extra=vol.ALLOW_EXTRA)
+
+
+@attr.s
+class ModemData:
+    """Class for modem state."""
+
+    modem = attr.ib()
+
+
+@attr.s
+class LTEData:
+    """Shared state."""
+
+    websession = attr.ib()
+    modem_data = attr.ib(init=False, factory=dict)
+
+    def get_modem_data(self, config):
+        """Get the requested or the only modem_data value."""
+        if CONF_HOST in config:
+            return self.modem_data.get(config[CONF_HOST])
+        if len(self.modem_data) == 1:
+            return next(iter(self.modem_data.values()))
+
+        return None
+
+
+async def async_setup(hass, config):
+    """Set up TP-Link LTE component."""
+    if DATA_KEY not in hass.data:
+        websession = async_create_clientsession(
```

Comment:
What's the reason that we need to create a new session and can't reuse the home assistant session? Is it the cookie?

Reply:
Yep, the authentication depends on the session id as cookie

Reply:
The issue is that the normal session does not allow cookie storage when the hostname is an IP address. We should probably make a generic solution for that, like we have for `verify_ssl`.